### ISSUE TITLE
Document how `poll_list` handles errors

### ISSUE THREAD

**Issue by sunfishcode** [WebAssembly/wasi-io#48]
The `poll_list` function itself doesn't have a return value so it's infallible. We should document how it behaves when passed invalid file descriptors.

**Comment by pchickey** [comment]
File descriptors being passed to poll_list are a thing of the past. It should be impossible to create an error because all Pollable resources are always valid to poll on, right?

**Comment by sunfishcode** [comment]
Yes, that's right. So we should document what happens if you're using bindings like C and you're using a handle after it's been dropped.

**Comment by pchickey** [comment]
Easy enough - it will trap, just like every other case of passing an invalid resource. Agreed we can put it in the docs.
